core/shared/src/main/scala/zio/ZTrace.scala

```diff
@@ -23,7 +23,11 @@ import scala.annotation.tailrec
 final case class ZTrace(
   fiberId: FiberId,
   stackTrace: Chunk[ZTraceElement]
-) {
+) { self =>
+
+  def ++(that: ZTrace): ZTrace =
```

Comment:
With this monoid defined for `ZTrace`, it would be great to have `Cause#addTrace`, which provides additive rather than replacement semantics. I think the existence of this `++` operator makes it well-defined.